__init__.py

```diff
@@ -472,3 +546,22 @@ def by_type(self, type: str) -> list[entity_instance]:
 
 def open(fn) -> file:
     return file(parse(filename=fn, with_tree=True, with_header=True))
+
+class file_description:
```

Comment:
These explicit classes seem to be not necessary when we have the dynamic approach with namespaces.

(collections.namedtuple would have been a 'nice' way btw to represent them, because then you can construct them directly from an iterable.)